### PR TITLE
fix target_platform for tests

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

cc @beckermr

### CODE REVIEW

**Review by beckermr — APPROVED** [review]

**Comment by beckermr** [comment]
oh fun!

```
Undefined symbols for architecture x86_64:
  "_quadmath_snprintf", referenced from:
      _write_complex in libgfortran.a(write.o)
      _write_float_0 in libgfortran.a(write.o)
      __gfortrani_write_real in libgfortran.a(write.o)
      __gfortrani_write_real_g0 in libgfortran.a(write.o)
  "_strtoflt128", referenced from:
      __gfortrani_convert_real in libgfortran.a(read.o)
      __gfortrani_convert_infnan in libgfortran.a(read.o)
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
```

**Comment by beckermr** [comment]
I think the build we just pushed is broken too. :(

**Comment by beckermr** [comment]
Do we need to add `-lomp` to the spec file or something?

```
Undefined symbols for architecture x86_64:
  "___emutls_get_address", referenced from:
      _gomp_resolve_num_threads in libgomp.a(parallel.o)
      _GOMP_parallel_start in libgomp.a(parallel.o)
      _GOMP_parallel_end in libgomp.a(parallel.o)
      _GOMP_parallel in libgomp.a(parallel.o)
      _GOMP_cancellation_point in libgomp.a(parallel.o)
      _GOMP_cancel in libgomp.a(parallel.o)
      _omp_get_num_threads in libgomp.a(parallel.o)
      ...
```

**Comment by beckermr** [comment]
also this one

```
Undefined symbols for architecture x86_64:
  "___emutls_get_address", referenced from:
      __gfortran_generate_error_common in libgfortran.a(error.o)
      __gfortran_generate_error in libgfortran.a(error.o)
      _async_io in libgfortran.a(async.o)
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
```

File: recipe/build.sh [hunk 1]
```diff
@@ -137,7 +137,7 @@ if [[ "$host_platform" == "$target_platform" ]]; then
   rm ${PREFIX}/lib/libgfortran.spec
   sed "s#@CONDA_PREFIX@#$PREFIX#g" $RECIPE_DIR/libgfortran.spec > ${PREFIX}/lib/libgfortran.spec
 
-  for file in libgfortran.spec libgomp.spec libgfortran.a libgomp.a; do
```
Comment (beckermr) [inline recipe/build.sh:140]: I guess gcc is smart enough to know that if it cannot find the `.a` file to link to the `.so`?

Comment (isuruf) [reply recipe/build.sh:140]: yes

**Pull request merged**